doc/system_requirements.md

```diff
@@ -0,0 +1,124 @@
+# System Requirement Specification Advanced Analytics Framework
+
+## Introduction
+
+This library is an event-driven framework that allows users to build complex 
+data processing algorithms in Exasol.
+
+
+## About This Document
+
+### Goal
+
+The goal of this library is to provide a general framework to implement complex 
+data analysis algorithms with Exasol. This framework provides certain features 
+to users in which they are able to run their implementations.
+
+## Stakeholders
+
+Software developers use this framework to develop complex data processing 
+algorithms such as machine learning methods.
+
+## Features
+
+Features are the highest level requirements in this document that describe the 
+main functionality of Advanced Analytics Framework.
+
+### Event Loop
+`feat~event-loop~1`
+
+Event Loop processes only the state transitions by executing queries returned by 
+the Event Handler. Some complex algorithms need a global loop. In each iteration 
+of the loop they need to execute SQL queries until the algorithm no longer 
+needs to run further queries. 
+
+Needs: req
+
+### Event Handler
+`feat~event-handler~1`
+
+TODO
+
+Needs: req
+
+
+## Functional Requirements
+
+This section lists functional requirements. The requirements are grouped by 
+feature where they belong to a single feature.
+
```

Comment:
We want to implement the algorithm in python, because it has a larger community and better tooling and other libs are already in this language